src/Generators/SubProfileGenerator.php

```diff
@@ -0,0 +1,120 @@
+   * The Lightning component discovery helper.
+   *
+   * @var \Drupal\lightning\ComponentDiscovery
+   */
+  protected $componentDiscovery;
+
+  /**
+   * Modules to include in the sub-profile.
+   *
+   * @var string[]
+   */
+  protected $installList = [];
+
+  protected $name = 'lightning-subprofile';
+  protected $description = 'Generates a Lightning Subprofile.';
+  protected $alias = 'lsp';
+  protected $templatePath = __DIR__;
+  protected $destination = 'profiles';
+
+  /**
+   * SubProfileGenerator constructor.
+   *
+   * @param string $app_root
+   *   The Drupal application root.
+   */
+  public function __construct($app_root) {
+    parent::__construct();
+    $this->componentDiscovery = new ComponentDiscovery($app_root);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function interact(InputInterface $input, OutputInterface $output)
+  {
+    $questions['name'] = new Question('Profile Name');
+    $questions['name']->setValidator([Utils::class, 'validateRequired']);
+
+    $questions['machine_name'] = new Question('Profile Machine Name (enter for default)');
+    $questions['machine_name']->setValidator([Utils::class, 'validateMachineName']);
+    $questions['description'] = new Question('Enter the description (optional)', NULL);
+    $questions['install'] = new Question('Additional modules to include (optional), separated by commas (e.g. context, rules, file_entity)', NULL);
+    $normalizer = function ($answer) {
+      return $answer ? array_map('trim', explode(',', $answer)) : [];
+    };
+    $questions['install']->setNormalizer($normalizer);
+
+    $questions['exclusions'] = new ConfirmationQuestion('Do you want to exclude any components of Lightning?', FALSE);
+
+
+    $vars = &$this->collectVars($input, $output, $questions);
+
+    if ($vars['exclusions']) {
+      $modules = $this->componentDiscovery->getAll();
+      $questions['exclude'] = new ChoiceQuestion(
+        'Lightning components to exclude (optional), entered as keys separated by commas (e.g. 0,1)',
+        array_keys($modules),
+        NULL
+      );
+      $questions['exclude']->setMultiselect(true);
+      $this->collectVars($input, $output, $questions);
+    }
+
+
+    $info_array = [
+      'name' => $vars['name'],
+      'type' => 'profile',
+      'description' => [],
+      'core_version_requirement' => '^8.8 || ^9',
+      'install'=> [],
+      'themes' => [
+        'bartik', 'seven'
+      ],
+      'base profile' => 'lightning',
+      'exclude'=> [],
+    ];
+
+    if ($vars['description']) {
+      $info_array['description'] = $vars['description'];
+    }
+    if ($vars['install']) {
+      $info_array['install'] = $vars['install'];
+    }
+    if ($vars['exclude']) {
+      $info_array['exclude'] = $vars['exclude'];
+    }
+
+    $info_array = array_filter($info_array);
+
+    $this->addFile()
+      ->path('custom/{machine_name}/{machine_name}.info.yml')
+      ->content(\Drupal\Component\Serialization\Yaml::encode($info_array));
```

Comment:
We should be `use` the Yaml class at the top of the file.

Reply:
oops :)